### PR TITLE
Fix/fixes

### PR DESCRIPTION
**Pull request by ElenaBitire**
## What does this do?

 - fix issues related to web support 
 - fix background color for Select's options 
 - update Button & Input variants for dark mode 

## How did you test this?

 - locally

### CODE REVIEW

File: src/ui/select.test.tsx [hunk 1]
```diff
@@ -1,4 +1,6 @@
 /* eslint-disable max-lines-per-function */
+import '@shopify/flash-list/jestSetup';
```
Comment (yjose) [inline src/ui/select.test.tsx:2]: I think it should be added to the test-util file?

File: src/ui/select.tsx [hunk 2]
```diff
@@ -91,14 +91,15 @@ export const Options = React.forwardRef<BottomSheetModal, OptionsProps>(
         index={0}
         snapPoints={snapPoints}
         backgroundStyle={{
-          backgroundColor: isDark ? colors.charcoal[800] : colors.white,
+          backgroundColor: isDark ? colors.neutral[800] : colors.white,
         }}
       >
-        <BottomSheetFlatList
+        <List
```
Comment (yjose) [inline src/ui/select.tsx:97]: @ElenaBitire **BottomSheetFlatList** is important here, as it is linked to the bottom sheet modal logic and can handle gestures much better than other lists.

I would suggest continuing to use BottomSheetFlatList and using FlatList on the web.

```js
// create a list component 
const List = Platform.OS === 'web' ? FlatList : BottomSheetFlatList 
// then you can use it, I suppose they have the same typing definition 
```

File: src/ui/select.tsx [hunk 3]
```diff
@@ -72,6 +75,9 @@ export const Options = React.forwardRef<BottomSheetModal, OptionsProps>(
     const snapPoints = React.useMemo(() => [height], [height]);
     const { colorScheme } = useColorScheme();
     const isDark = colorScheme === 'dark';
+
+    const List = Platform.OS === 'web' ? FlashList : BottomSheetFlatList;
```
Comment (yjose) [inline src/ui/select.tsx:79]: Should be outside the component

**Review by yjose — CHANGES_REQUESTED** [review]

**Comment by hamzamekk** [comment]
is Nativewind 4 the source of issue in Windows ?

Upon encountering the issue mentioned in the GitHub [link](https://github.com/obytes/react-native-template-obytes/issues/289), which was reported by a user facing problems on Windows, I realized I could replicate the scenario since I also have a Windows system. To address the issue, I embarked on a debugging journey. However, the absence of clear error messages made it challenging to pinpoint the root cause, especially considering our template's complex library dependencies.

To investigate further, I conducted a test by creating two projects—one with the Expo router and another without. Surprisingly, both projects functioned smoothly, particularly on the web platform. Encouraged by this, I decided to integrate nativewind version 4 into both projects, following the [provided instructions](https://www.nativewind.dev/v4/getting-started/react-native).

Unexpectedly, upon integration, both projects encountered the same issue observed in our template—getting stuck on the terminal during the bundling process. This recurrence strongly suggested that nativewind 4 might be the underlying cause of the problem.

In an attempt to validate this hypothesis, I delved deeper into research and came across discussions indicating potential issues with the Metro config file. However, removing it did run all three projects but not really sure what's going on behind the scene and for our template it didn't really run due to compatibility with web for several libs.

In conclusion, the problem seems to be closely associated with the configuration of nativewind 4.

**Comment by yjose** [comment]
Thank you @hamzamekk for your time, can you please check if this fix https://github.com/marklawlor/nativewind/pull/854 is working as expected

**Comment by hamzamekk** [comment]
> Thank you @hamzamekk for your time, can you please check if this fix [marklawlor/nativewind#854](https://github.com/marklawlor/nativewind/pull/854) is working as expected

you're welcome, and for the solution it's working perfectly

**Comment by yjose** [comment]
Thank you @hamzamekk 🙏
@ElenaBitire  We will wait until the PR is merged to NativeWind, and then we can merge this PR as well.

**Comment by ElenaBitire** [comment]
> Thank you @hamzamekk 🙏 @ElenaBitire We will wait until the PR is merged to NativeWind, and then we can merge this PR as well.

Sure @yjose and thank you very much @hamzamekk for taking your time! 🙏

**Comment by hamzamekk** [comment]
Thank you @yjose @ElenaBitire my pleasure to help 👌👌

**Comment by dmitryshostak** [comment]
I've mentioned the nativewind's creator to merge the fix ASAP. https://github.com/marklawlor/nativewind/pull/860 Hoping he'll respond soon. Blocking us a little as well.

Btw, it's the first time I see `react-native-template-obytes` - splendid job, guys! Your stack is almost identical to ours 👍 ❤️

**Comment by tonven** [comment]
Hi. I noticed that this line is commented out in your template: https://github.com/obytes/react-native-template-obytes/blob/b894d276ad9a03d82fcee8ddc0bf8a365aa96d27/src/core/hooks/use-selected-theme.tsx#L36
So this means it will not load current theme from mmkv on app restart / web refresh. Is it supposed to be like this? Thanks :smile:

**Comment by yjose** [comment]
Nice catch @tonven , I remember commenting on this line as the function was not exported correctly from nativewind. I opened a PR to Nativewind to fix this and it was already merged, i updated it in our internal projects but forgot to update it here

**Comment by ElenaBitire** [comment]
We are going to merge this PR since it contains other fixes but the [issue](https://github.com/obytes/react-native-template-obytes/issues/289) related to web support on Windows system remains open.
